armi/reactor/blocks.py

```diff
@@ -2414,14 +2407,21 @@ def getHydraulicDiameter(self):
         r"""
         Return the hydraulic diameter in this block in cm.
 
-        Hydraulic diameter is 4A/P where A is the flow area and P is the wetted perimeter.
-        In a hex assembly, the wetted perimeter includes the cladding, the wire wrap, and the
-        inside of the duct. The flow area is the inner area of the duct minus the area of the
-        pins and the wire.
+        Hydraulic diameter is 4A/P where A is the flow area and P is the wetted perimeter. In a hex
+        assembly, the wetted perimeter includes the cladding, the wire wrap, and the inside of the
+        duct. The flow area is the inner area of the duct minus the area of the pins and the wire.
         """
         return 4.0 * self.getFlowArea() / self.getWettedPerimeter()
 
 
+class FilletedHexBlock(HexBlock):
+    """Defines a Block shaped like a hexagon with rounded corners."""
+
+    PITCH_COMPONENT_TYPE: ClassVar[_PitchDefiningComponent] = (
+        components.FilletedHexagon,
+    )
+
+
```

Comment:
Is a block with rounded corners something that is wanted? This is confusing to me, because `Block`s are used to define the core lattice, so what is then happening to the empty space between blocks near the rounded corners?

Reply:
So, your notion is that we should have `FilletedHexagon` as a shape for Components, but not have a Block of with that shape?

@dlangewisch Will having just the "rounded hexagon" _shape_ be enough for you to build components of this shape? Or do you need entire blocks with this shape?

Reply:
Correct, that is what I would expect. I do not know the full motivation for having this new shape, but it doesn't seem to make sense to have it define a block.